Makefile

```diff
@@ -8,268 +28,405 @@ usage:
 	@echo
 	@echo "The next step is to choose a board to build Tock for. Mainline"
 	@echo "Tock currently includes support for the following platforms:"
-	@for f in `./tools/list_boards.sh`; do printf " - $$f\n"; done
+	@for f in $(ALL_BOARDS); do printf " - $$f\n"; done
 	@echo
 	@echo "Run 'make' in a board directory to build Tock for that board,"
 	@echo "and usually 'make program' or 'make flash' to load Tock onto hardware."
 	@echo "Check out the README in your board's folder for more information."
 	@echo
 	@echo "This root Makefile has a few useful targets as well:"
-	@echo "        allboards: Compiles Tock for all supported boards"
+	@echo "        allaudit: Audit Cargo dependencies for all kernel sources"
+	@echo "       allboards: Compiles Tock for all supported boards"
 	@echo "        allcheck: Checks, but does not compile, Tock for all supported boards"
 	@echo "          alldoc: Builds Tock documentation for all boards"
-	@echo "           audit: Audit Cargo dependencies for all kernel sources"
-	@echo "              ci: Run all continuous integration tests"
 	@echo "           clean: Clean all builds"
-	@echo "          clippy: Runs the clippy code linter with Tock's default arguments"
-	@echo " emulation-check: Run the emulation tests for supported boards"
-	@echo " emulation-setup: Setup QEMU for the emulation tests"
 	@echo "          format: Runs the rustfmt tool on all kernel sources"
-	@echo "    format-check: Checks if the rustfmt tool would require changes, but doesn't make them"
-	@echo "            list: Lists available boards"
```

Comment:
`format-check` is the exact same as `ci-job-format` and doesn't seem like a useful alias

`make list` basically prints the boards directory (caveat nordic for now) and a plain `make` also lists all of the boards already anyway. Nothing was using it, and we have the `list_boards` in tools if something elsewhere needs it. Was trying to reduce clutter in the options presented to users and this doesn't seem useful.

Reply:
I was thinking of make list, which is a user-facing API, so it's not too surprising that nothing in the tree is using it. I used it with tinyos all the time, which is why I added it here a few years ago. Also, we reference it in the book.

Reply:
Hmm.. thinking on this more, I wonder if this isn't a bit of an anti-pattern. I think part of my takeaway from #1784 was that we should push towards the `boards` folder just being `boards`, and it shouldn't require tooling to figure what boards exist in the ecosystem.

I'm okay with adding this back in the short term if you think it's necessary, but I think once we've finished component-izing the nordic boards and re-org'd that directory to be a flat namespace of boards, this should go away (and the tutorial can say `ls boards/` to see supported boards).

Reply:
In my opinion the point of supporting `make` is it just works and is developer and new user friendly. I don't see a reason to shift away from that thinking. If we can make any step easier for users, who maybe don't know to just switch to `/boards`, why not? What do we lose? The Makefile is already 435 lines, why not spend more of those on providing guiding hooks for new, unfamiliar users? I suppose if it is a burden to maintain it might not be worth it, but this has been working for the last four years I think.

Reply:
It's nothing to do with the complexity of the Makefile, it's trying to make things easier for new developers by presenting _fewer_ options. Right now running just `make` dumps out 40 lines of text and 12 possible commands. My eyes certainly tend to glaze over when I run `random_command -h` and it dumps a page of text. The goal is to reduce the number of commands that new users have to learn.

What value is `make list` if all it does is run `ls boards/`?

Reply:
It doesn't!

```
Supported Tock Boards:
 - nordic/nrf52dk
 - nordic/nrf52840_dongle
 - nordic/nrf52840dk
 - opentitan
 - stm32f3discovery
 - redboard_artemis_nano
 - launchxl
 - hifive1
 - hail
 - arty_e21
 - nucleo_f429zi
 - acd52832
 - imix
 - nucleo_f446re

To build the kernel for a particular board, change to that directory
and run make:
    cd boards/hail
    make
```

is not

```
ls boards/
Makefile.common		hail			nano33ble		redboard_artemis_nano
README.md		hifive1			nordic			stm32f3discovery
acd52832		imix			nucleo_f429zi
arty_e21		kernel_layout.ld	nucleo_f446re
components		launchxl		opentitan
```

One provides context and help, plus much better formatting, and doesn't include things that are not boards. If it essential to remove things from the printout, why does a new user need to know how CI is implemented? Or how to build the documentation?

Reply:
All right, you've convinced me. I've added it back.